spec/client_spec.rb

```diff
@@ -92,6 +92,22 @@
         expect(lock_manager).to receive(:sleep).exactly(lock_manager_opts[:retry_count] - 1).times
         lock_manager.lock(resource_key, ttl)
       end
+
+      it 'sleeps at least the specified retry_delay in milliseconds' do
```

Comment:
Yeah, that would be the way I'd usually test stuff like this. Stub out random/external factors, then test that the calculation is done correctly. If you then would want to be super-ambitious about proper testing, you'd need to cover all edge cases of possible values you stubbed out (so here for `rand` probably 0, max-1, and somewhere in the middle).

But as said, I don't think it's super important here, and I'm very confident these tests will never fail :+1: